src/components/tools/geometry-tool/use-label-segment-dialog.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from "react";
+import LineLabelSvg from "../../../clue/assets/icons/geometry/line-label.svg";
+import { ESegmentLabelOption } from "../../../models/tools/geometry/jxg-changes";
+import { getPolygonEdge } from "../../../models/tools/geometry/jxg-polygon";
+import { useCustomModal } from "../../../hooks/use-custom-modal";
+import "./label-segment-dialog.sass";
+
+interface LabelRadioButtonProps {
+  display: string;
+  label: string;
+  checkedLabel: string;
+  setLabelOption: React.Dispatch<React.SetStateAction<string>>;
+}
+const LabelRadioButton: React.FC<LabelRadioButtonProps> = ({display, label, checkedLabel, setLabelOption}) => {
+  return (
+    <label className="radio-button-label">
+      <input
+        className="radio-button"
+        type="radio"
+        id={label}
+        name="labelOption"
+        value={label}
+        checked={label === checkedLabel}
+        onChange={e => {
+          if (e.target.checked) {
+            setLabelOption(e.target.value);
+          }
+        }}
+      />
+      {display}
+    </label>
+  );
+};
+
+interface IContentProps {
+  labelOption: string;
+  setLabelOption: React.Dispatch<React.SetStateAction<string>>;
+}
+const Content: React.FC<IContentProps> = ({ labelOption, setLabelOption }) => {
+  return (
+    <fieldset className="radio-button-set">
+      <LabelRadioButton
+        display="None"
+        label={ESegmentLabelOption.kNone}
+        checkedLabel={labelOption}
+        setLabelOption={setLabelOption}
+      />
+      <LabelRadioButton
+        display="Label"
+        label={ESegmentLabelOption.kLabel}
+        checkedLabel={labelOption}
+        setLabelOption={setLabelOption}
+      />
+      <LabelRadioButton
+        display="Length"
+        label={ESegmentLabelOption.kLength}
+        checkedLabel={labelOption}
+        setLabelOption={setLabelOption}
+      />
+    </fieldset>
+  );
+};
+
+function getPolygonSegment(board: JXG.Board, polygon: JXG.Polygon, points: [JXG.Point, JXG.Point]) {
+  const pointIds = points.map(pt => pt.id);
+  return getPolygonEdge(board, polygon.id, pointIds);
+}
+
+interface IProps {
+  board: JXG.Board;
+  polygon: JXG.Polygon;
+  points: [JXG.Point, JXG.Point];
+  onAccept: (polygon: JXG.Polygon, points: [JXG.Point, JXG.Point], labelOption: ESegmentLabelOption) => void;
+  onClose: () => void;
+}
+export const useLabelSegmentDialog = ({ board, polygon, points, onAccept, onClose }: IProps) => {
+  // TODO: Do we need to find the segment on every render?
+  const segment = getPolygonSegment(board, polygon, points);
```

Comment:
Presumably not -- wrapping the call in `useMemo()` would make sure that it is only called when necessary.